libbeat/autodiscover/builder/helper.go

```diff
@@ -155,41 +156,21 @@ func IsNoOp(hints common.MapStr, key string) bool {
 }
 
 // GenerateHints parses annotations based on a prefix and sets up hints that can be picked up by individual Beats.
-func GenerateHints(annotations common.MapStr, container, prefix string) common.MapStr {
+func GenerateHints(annotations common.MapStr, container, prefix string, separator string) common.MapStr {
+	validPrefix := regexp.MustCompile("^(?P<Module>[a-zA-Z0-9]*)\\.?(" + container + ")?" + separator + "(?P<Key>.*)$")
```

Comment:
Separator here should be converted to valid regexp, as done in the test by writing the `.`  as `\\.`